selfdrive/debug/internal/qlog_size.py

```diff
@@ -48,9 +49,26 @@ def make_pie(msgs, typ):
 if __name__ == "__main__":
   parser = argparse.ArgumentParser(description='View log size breakdown by message type')
   parser.add_argument('route', help='route to use')
+  parser.add_argument('--as-qlog', action='store_true', help='decimate rlog using latest decimation factors')
   args = parser.parse_args()
 
   msgs = list(LogReader(args.route))
 
+  if args.as_qlog:
+    new_msgs = []
+    msg_cnts = defaultdict(int)
+    for msg in msgs:
+      msg_which = msg.which()
+      if msg.which() in ("initData", "sentinel"):
+        new_msgs.append(msg)
+        continue
+
+      decimation = SERVICE_LIST[msg_which].decimation
```

Comment:
Add check whether message type exists in SERVICE_LIST, old qlogs may have deprecated messages and it will crash